src/Features/Core/Portable/CustomMessageHandler/RegisterHandlersResponse.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.Serialization;
+
+namespace Microsoft.CodeAnalysis.CustomMessageHandler;
+
+[DataContract]
+internal readonly struct RegisterHandlersResponse(string[] handlers, string[] documentHandlers)
```

Comment:
can just be a readonly record string, with attributes on the parameter properties.